src/kits/communityWebServer/StaticFileWeblet.sedona

```diff
@@ -0,0 +1,273 @@
+//
+// History:
+//   30 Jun 18  Vincent Wang  Static File Weblet
+//
+
+** 
+** StaticFileWeblet provides the ability to serve static files(html, css, js,
+** images etc) through http, so that user can use sedona as a minimal web
+** server and host a static web app(SPA for example) just in sedona, without 
+** any outside dependency. 
+**
+** To enable this weblet, you need to create an object of this component as a 
+** child of WebService. Deleting the object will disable the weblet. This is
+** different comparing to SpyWeblet. 
+**
+** You can create multiple objects of StaticFileWeblet and every object uses
+** different urlPrefix, so that sedona is possible to host multiple web apps.
+**
+** To save space on sedona, gzipped file is supported and I recommend to use it
+** since that will be more efficient(less bytes to be loaded and transferred)
+** 
+class StaticFileWeblet extends Weblet
+{
+  @config @asStr property Buf(12) urlPrefix = "statics"
+////////////////////////////////////////////////////////////////
+// Weblet Registration
+////////////////////////////////////////////////////////////////
+
+  override void start() 
+  {
+    register()
+  }
+
+  override void stop()
+  {
+    unregister()
+  }
+
+////////////////////////////////////////////////////////////////
+// Weblet
+////////////////////////////////////////////////////////////////
+
+  ** 
+  ** all static files to be served must be put under 'statics' folder
+  ** and the url will start with 'statics', too. If you want to use other
+  ** name, config it by urlPrefix slot, Note: the max length is 11 chars
+  **
+  override Str prefix() { return urlPrefix.toStr() }
+
+  override Str description() { return "Sedona Static File Weblet" }
+
+  **
+  ** since we only serve static file here, so only 'GET' method is supported
+  ** 
+  override void service(WebReq req, WebRes res)
+  {
+    Str method = req.method;
+    if (method.equals("GET")) 
+      get(req, res)
+    else {
+      res.writeStatus(HttpCode.methodNotAllowed)
+      res.writeHeader("Allow", "GET")
+    }
+  }
+
+  ** 
+  ** handle 'GET' request, try to find the static file and return it 
+  **
+  override void get(WebReq req, WebRes res)
+  {
+    reset()
+
+    prepareFile(req)
+
+    if (isDir(filePath)) {
+      tryAppendIndex(filePath)
+    }
+
+    serveFile(res)
+  }
+
+  internal void reset() 
+  {
+    filePath.set(0, 0)
+  }
+
+  internal void serveFile(WebRes res) 
+  {
+    if (file.name==null) {
+      res.writeStatus(HttpCode.notFound).finishHeaders()
+      return
+    }
+
+    if (!file.exists()) {
+      //try gzipped content
+      appendStr(file.name, ".gz", pathStrLen)
+
+      if (!file.exists()) {
+        res.writeStatus(HttpCode.notFound).finishHeaders()
+        return
+      }
+    }
+
+    if (!file.open("r")) {
+      res.writeStatus(HttpCode.internalError).finishHeaders()
+      return
+    }
+
+    App.log.message("Serving File: " + file.name);
```

Comment:
Two fixes needed here:
- the log object from the WebService should be referred here, not the App
- use trace instead of message, otherwise, it'll flood the logs on small devices during runtime

Reply:
This has been fixed.